static/app/views/settings/projectSourceMaps/sourceMapsDetails.tsx

```diff
@@ -158,6 +158,47 @@ export function SourceMapsDetails({params, location, router, project}: Props) {
     }
   );
 
+  const releaseVersions = debugIdBundlesArtifacts?.associations.map(
+    association => `"${association.release}"`
+  );
+
+  const {data: releasesData, isPending: releasesLoading} = useApiQuery<Release[]>(
+    [
+      `/organizations/${organization.slug}/releases/`,
+      {
+        query: {
+          project: [project.id],
+          query: releaseVersions ? `release:[${releaseVersions.join(',')}]` : undefined,
+        },
+      },
+    ],
+    {
+      staleTime: Infinity,
+      retry: false,
+      enabled: isDebugIdBundle && !debugIdBundlesArtifactsLoading,
+    }
+  );
+
+  const debugIdBundlesArtifactsData = useMemo(() => {
+    if (releasesLoading) {
+      return debugIdBundlesArtifacts;
+    }
+
+    if (!debugIdBundlesArtifacts) {
+      return undefined;
+    }
+
+    const existingReleaseNames = new Set((releasesData ?? []).map(r => r.version));
+
+    return {
+      ...debugIdBundlesArtifacts,
+      associations: debugIdBundlesArtifacts.associations.map(association => ({
+        ...association,
+        exists: existingReleaseNames.has(association.release),
+      })),
+    };
+  }, [releasesLoading, releasesData, debugIdBundlesArtifacts]);
```

Comment:
We can probably extract this into a reusable function, but let's clean that up in a later pass